### PR TITLE
CLI: Integrate control-plane actions #1828

### PR DESCRIPTION
**Pull request by rakeshsharma14317**

### CODE REVIEW

File: cli/cmd/project/create.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+			}
+
+			fmt.Printf("Created project: %v\n", proj)
+			sp.Stop()
+			return nil
+		},
+	}
+
+	createCmd.Flags().SortFlags = false
+	createCmd.Flags().StringVar(&displayName, "display-name", "noname", "Display name")
```
Comment (begelundmuller) [inline cli/cmd/project/create.go:49]: The default display name should not be "noname". It should be empty in that case.

Also, it's a description field, not a display name field.

File: cli/pkg/config/config.go [hunk 2]
```diff
@@ -3,10 +3,11 @@ package config
...
 
 type Config struct {
-	Version    Version
-	AdminURL   string
-	AdminToken string
-	DefaultOrg string
+	Version             Version
+	AdminURL            string
+	OverridenAdminToken string
+	DefaultAdminToken   string
```
Comment (begelundmuller) [inline cli/pkg/config/config.go:9]: Make them `AdminTokenDefault` and `AdminTokenOverride`

File: cli/pkg/config/config.go [hunk 3]
```diff
@@ -29,3 +30,14 @@ func (v Version) String() string {
 func (v Version) IsDev() bool {
 	return v.Number == ""
 }
+
+func (c *Config) IsAuthenticated() bool {
+	return (c.GetAdminToken() != "" && c.AdminURL != "")
+}
```
Comment (begelundmuller) [inline cli/pkg/config/config.go:36]: It doesn't have to check that AdminURL is empty

File: cli/pkg/config/config.go [hunk 4]
```diff
@@ -29,3 +30,14 @@ func (v Version) String() string {
...
+
+func (c *Config) GetAdminToken() string {
```
Comment (begelundmuller) [inline cli/pkg/config/config.go:38]: Just call it `AdminToken()` (remove the `Get`)

**Review by begelundmuller — CHANGES_REQUESTED** [review]
It's also still missing pretty printing of objects in tables

File: cli/cmd/org/show.go [hunk 5]
```diff
@@ -16,6 +16,9 @@ func ShowCmd(cfg *config.Config) *cobra.Command {
 		Short: "Show",
 		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
+			sp := cmdutil.Spinner("Switching org...")
```
Comment (begelundmuller) [inline cli/cmd/org/show.go:19]: Wrong message

File: cli/cmd/org/org.go [hunk 6]
```diff
@@ -18,6 +21,32 @@ func OrgCmd(cfg *config.Config) *cobra.Command {
 	orgCmd.AddCommand(InviteCmd(cfg))
 	orgCmd.AddCommand(MembersCmd(cfg))
 	orgCmd.AddCommand(SwitchCmd(cfg))
+	orgCmd.AddCommand(ListCmd(cfg))
+	orgCmd.AddCommand(DeleteCmd(cfg))
 
 	return orgCmd
 }
+
+func toOrgs(organizations []*adminv1.Organization) []*organization {
```
Comment (begelundmuller) [inline cli/cmd/org/org.go:30]: call `toTable`

File: cli/cmd/org/org.go [hunk 7]
```diff
@@ -18,6 +21,32 @@ func OrgCmd(cfg *config.Config) *cobra.Command {
...
+	orgs := make([]*organization, 0, len(organizations))
+
+	for _, org := range organizations {
+		orgs = append(orgs, toOrg(org))
+	}
+
+	return orgs
+}
+
+func toOrg(o *adminv1.Organization) *organization {
```
Comment (begelundmuller) [inline cli/cmd/org/org.go:40]: call `toRow`

File: cli/cmd/org/switch.go [hunk 8]
```diff
@@ -3,16 +3,34 @@ package org
...
-			fmt.Println("not implemented")
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			sp := cmdutil.Spinner("Switching org...")
+			sp.Start()
+
+			_, err := dotrill.GetDefaultOrg()
+			if err != nil {
+				return err
+			}
```
Comment (begelundmuller) [inline cli/cmd/org/switch.go:24]: Why is this needed?

File: cli/cmd/org/switch.go [hunk 9]
```diff
@@ -3,16 +3,34 @@ package org
...
+
+			err = dotrill.SetDefaultOrg(args[0])
+			if err != nil {
+				return err
+			}
```
Comment (begelundmuller) [inline cli/cmd/org/switch.go:29]: We should check the org exists before doing this

File: cli/cmd/org/create.go [hunk 10]
```diff
@@ -26,18 +29,20 @@ func CreateCmd(cfg *config.Config) *cobra.Command {
 
 			org, err := client.CreateOrganization(context.Background(), &adminv1.CreateOrganizationRequest{
 				Name:        args[0],
-				Description: displayName,
+				Description: description,
 			})
```
Comment (begelundmuller) [inline cli/cmd/org/create.go:33]: After this, we should `switch` to it

File: cli/cmd/project/create.go [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+)
+
+func CreateCmd(cfg *config.Config) *cobra.Command {
+	var description string
+	createCmd := &cobra.Command{
+		Use:   "create",
+		Args:  cobra.ExactArgs(1),
+		Short: "Create",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			sp := cmdutil.Spinner("Creating project...")
```
Comment (begelundmuller) [inline cli/cmd/project/create.go:20]: We should remove this command – people use `connect` to setup a project

File: cli/cmd/project/edit.go [hunk 12]
```diff
@@ -1,29 +1,54 @@
...
+			sp := cmdutil.Spinner("Updating project...")
+			sp.Start()
+
+			client, err := client.New(cfg.AdminURL, cfg.AdminToken())
+			if err != nil {
+				return err
+			}
+			defer client.Close()
+
+			// Todo how will get the org name? will it be flag with cmd.
```
Comment (begelundmuller) [inline cli/cmd/project/edit.go:31]: It should use the default org

File: cli/cmd/project/list.go [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	"github.com/spf13/cobra"
+)
+
+func ListCmd(cfg *config.Config) *cobra.Command {
+	listCmd := &cobra.Command{
+		Use:   "list",
+		Short: "List",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			sp := cmdutil.Spinner("Listing project...")
```
Comment (begelundmuller) [inline cli/cmd/project/list.go:18]: Should be `Listing projects...`

File: cli/cmd/project/project.go [hunk 14]
```diff
@@ -1,21 +1,50 @@
...
 	projectCmd.AddCommand(StatusCmd(cfg))
 	projectCmd.AddCommand(ConnectCmd(cfg))
 	projectCmd.AddCommand(EditCmd(cfg))
 	projectCmd.AddCommand(DeleteCmd(cfg))
+	projectCmd.AddCommand(ListCmd(cfg))
 
 	return projectCmd
 }
+
+func toProjects(projects []*adminv1.Project) []*project {
```
Comment (begelundmuller) [inline cli/cmd/project/project.go:28]: `toTable`

File: cli/cmd/project/project.go [hunk 15]
```diff
@@ -1,21 +1,50 @@
...
+	orgs := make([]*project, 0, len(projects))
+
+	for _, org := range projects {
+		orgs = append(orgs, toProject(org))
+	}
+
+	return orgs
+}
+
+func toProject(o *adminv1.Project) *project {
```
Comment (begelundmuller) [inline cli/cmd/project/project.go:38]: `toRow`

**Review by begelundmuller — CHANGES_REQUESTED** [review]

File: cli/cmd/project/edit.go (hunk 12 above)
Comment (rakeshsharma14317) [reply cli/cmd/project/edit.go:31]: yeah, added this comment when getDefault org was not there :)

**Review by begelundmuller — APPROVED** [review]
